### ISSUE TITLE
Many-to-many relationships and Core Data

### ISSUE THREAD

**Issue by pedrommcarrasco** [mentrena/SyncKit#174]
Are many-to-many relationships on Core Data  still unsupported? 🤔

**Comment by mentrena** [comment]
Correct, CloudKit doesn't support them. CloudKit only supports many-to-one relationships by using a reference in each of the "many" records to the "one" record (see: https://developer.apple.com/documentation/cloudkit/ckrecord/reference)

**Comment by pedrommcarrasco** [comment]
Out of curiosity, does `NSPersistentCloudKitContainer` support them? - didn't see much information around 🤔

**Comment by mentrena** [comment]
That's a good question, the most I found is [this](https://developer.apple.com/documentation/coredata/mirroring_a_core_data_store_with_cloudkit/creating_a_core_data_model_for_cloudkit):
```
All relationships must be optional. Due to operation size limitations, CloudKit may not save relationship changes atomically.
All relationships must have an inverse, in case the records synchronize out of order.
CloudKit doesn’t support the Deny deletion rule.
```

which doesn't specifically say it doesn't support many-to-many, although in my head inverse relationships don't make sense in the case of many-to-many relationships. But I haven't tried it myself –nor have I really used `NSPersistentCloudKitContainer`.

Actually, after reading the docs for [CKRecord](https://developer.apple.com/documentation/cloudkit/ckrecord) again I see properties can be arrays of references, so I think it should be possible to support many-to-many relationships. I'll have a try at adding support as time allows :)
